app/code/Magento/ConfigurableProductGraphQl/Model/Resolver/Options.php

```diff
@@ -27,52 +25,37 @@ class Options implements ResolverInterface
      */
     private $optionCollection;
 
-    /**
-     * @var ValueFactory
-     */
-    private $valueFactory;
-
     /**
      * @var MetadataPool
      */
     private $metadataPool;
 
     /**
      * @param OptionCollection $optionCollection
-     * @param ValueFactory $valueFactory
      * @param MetadataPool $metadataPool
      */
     public function __construct(
         OptionCollection $optionCollection,
-        ValueFactory $valueFactory,
         MetadataPool $metadataPool
     ) {
         $this->optionCollection = $optionCollection;
-        $this->valueFactory = $valueFactory;
         $this->metadataPool = $metadataPool;
     }
 
     /**
      * Fetch and format configurable variants.
      *
-     * {@inheritDoc}
+     * {@inheritdoc}
      */
-    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null) : Value
+    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null)
     {
         $linkField = $this->metadataPool->getMetadata(ProductInterface::class)->getLinkField();
         if ($value['type_id'] !== Type::TYPE_CODE || !isset($value[$linkField])) {
-            $result = function () {
-                return null;
-            };
-            return $this->valueFactory->create($result);
+            return null;
         }
 
         $this->optionCollection->addProductId((int)$value[$linkField]);
 
-        $result = function () use ($value, $linkField) {
-            return $this->optionCollection->getAttributesByProductId((int)$value[$linkField]);
-        };
-
-        return $this->valueFactory->create($result);
+        return $this->optionCollection->getAttributesByProductId((int)$value[$linkField]);
```

Comment:
This was not supposed to be replaced because this was actual performance optimization via Deferred mechanism.